pkg/karmadactl/promote/promote.go

```diff
@@ -249,10 +268,238 @@ func (o *CommandPromoteOption) Run(f util.Factory, args []string) error {
 		return fmt.Errorf("failed to get gvr from %q: %v", o.gvk, err)
 	}
 
-	return o.promote(controlPlaneRestConfig, obj, gvr)
+	if o.Deps {
+		err := o.promoteDeps(memberClusterFactory, obj, mapper, gvr, controlPlaneRestConfig)
+		if err != nil {
```

Comment:
Continue to promote, any considerations here? Why do it like this?

Reply:
fixed